src/rpc/xmlrpc_tinyxml2.cc

```diff
@@ -0,0 +1,454 @@
+// (at your option) any later version.
+// 
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+// 
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+//
+// In addition, as a special exception, the copyright holders give
+// permission to link the code of portions of this program with the
+// OpenSSL library under certain conditions as described in each
+// individual source file, and distribute linked combinations
+// including the two.
+//
+// You must obey the GNU General Public License in all respects for
+// all of the code used other than OpenSSL.  If you modify file(s)
+// with this exception, you may extend this exception to your version
+// of the file(s), but you are not obligated to do so.  If you do not
+// wish to do so, delete this exception statement from your version.
+// If you delete this exception statement from all source files in the
+// program, then also delete it here.
+//
+// Contact:  Jari Sundell <jaris@ifi.uio.no>
+//
+//           Skomakerveien 33
+//           3185 Skoppum, NORWAY
+
+#include "config.h"
+
+#ifdef HAVE_XMLRPC_TINYXML2
+#ifndef HAVE_XMLRPC_C
+
+#include <cctype>
+#include <string>
+
+#include <stdlib.h>
+
+#include <rak/string_manip.h>
+#include <torrent/object.h>
+#include <torrent/exceptions.h>
+
+#include "vendor/tinyxml2/tinyxml2.h"
+#include "utils/base64.h"
+#include "xmlrpc.h"
+#include "parse_commands.h"
+
+namespace rpc {
+
+// Taken from xmlrpc-c
+const int XMLRPC_INTERNAL_ERROR = -500;
+const int XMLRPC_TYPE_ERROR = -501;
+const int XMLRPC_INDEX_ERROR = -502;
+const int XMLRPC_PARSE_ERROR = -503;
+const int XMLRPC_NETWORK_ERROR = -504;
+const int XMLRPC_TIMEOUT_ERROR = -505;
+const int XMLRPC_NO_SUCH_METHOD_ERROR = -506;
+const int XMLRPC_REQUEST_REFUSED_ERROR = -507;
+const int XMLRPC_INTROSPECTION_DISABLED_ERROR = -508;
+const int XMLRPC_LIMIT_EXCEEDED_ERROR = -509;
+const int XMLRPC_INVALID_UTF8_ERROR = -510;
+
+class xmlrpc_error : public torrent::base_error {
+public:
+  xmlrpc_error(int type, std::string msg) : m_type(type), m_msg(msg) {}
+  virtual ~xmlrpc_error() throw() {}
+
+  virtual int         type() const throw() { return m_type; }
+  virtual const char* what() const throw() { return m_msg.c_str(); }
+
+private:
+  int                 m_type;
+  std::string         m_msg;
+};
+
+const tinyxml2::XMLElement*
+element_access(const tinyxml2::XMLElement* elem, std::string elemNames) {
+  // Helper function to check each step of a element access, in lieu of XPath
+  char token = ',';
+  std::string stack("");
+  const tinyxml2::XMLElement* result = elem;
+  size_t start = 0;
+  size_t end = 0;
```

Comment:
Rewrite to only use `size_t pos = 0;`.

Reply:
FWIW, did a quick benchmark and it optimizes out the same. Changed for consistent style though.

Reply:
It's not a benchmark thing, it's for code clarity and proper location of variables inside/outside the while block.